src/utils/tokens.ts

```diff
@@ -361,6 +364,28 @@ const tokens = {
       });
     }
 
+    if (data !== null) {
```

Comment:
suggestion(non-blocking): Can you move this to the if in the line 310 (reordering whatever is needed)?

Reply:
The expected behavior is that data outputs should be at the end (unless it is an NFT).
We have some tests that expect this behavior.

I could move but then any additional output would need to be spliced and this behavior changes with `unshiftData`.
This makes the code simpler in my opinion, what do you think?